aeron-client/src/main/c/aeron_client_conductor.c

```diff
@@ -1634,11 +1719,24 @@ int aeron_client_conductor_async_add_counter(
 
     *async = NULL;
 
-    if (aeron_alloc((void **)&cmd, sizeof(aeron_async_add_counter_t)) < 0 ||
-        aeron_alloc((void **)&key_buffer_copy, key_buffer_length) < 0 ||
-        aeron_alloc((void **)&label_buffer_copy, label_buffer_length + 1) < 0)
+    if (aeron_alloc((void **)&cmd, sizeof(aeron_async_add_counter_t)) < 0)
+    {
+        AERON_APPEND_ERR("%s", "Unable to allocate counter");
+        return -1;
+    }
+
+    if (aeron_alloc((void **)&key_buffer_copy, key_buffer_length) < 0)
+    {
+        AERON_APPEND_ERR("%s", "Unable to allocate counter");
+        aeron_free(cmd);
+        return -1;
+    }
+
+    if (aeron_alloc((void **)&label_buffer_copy, label_buffer_length + 1) < 0)
     {
         AERON_APPEND_ERR("%s", "Unable to allocate counter");
+        aeron_free(cmd);
+        aeron_free(key_buffer_copy);
```

Comment:
Would a `goto` with a clean up section better in case future changes get missed?

Reply:
Done.